client/v3/retry_interceptor.go

```diff
@@ -109,14 +109,10 @@ func (c *Client) streamClientInterceptor(optFuncs ...retryOption) grpc.StreamCli
 	return func(ctx context.Context, desc *grpc.StreamDesc, cc *grpc.ClientConn, method string, streamer grpc.Streamer, opts ...grpc.CallOption) (grpc.ClientStream, error) {
 		ctx = withVersion(ctx)
 		// getToken automatically
-		// TODO(cfc4n): keep this code block, remove codes about getToken in client.go after pr #12165 merged.
-		if c.authTokenBundle != nil {
-			// equal to c.Username != "" && c.Password != ""
-			err := c.getToken(ctx)
-			if err != nil && rpctypes.Error(err) != rpctypes.ErrAuthNotEnabled {
-				c.GetLogger().Error("clientv3/retry_interceptor: getToken failed", zap.Error(err))
-				return nil, err
-			}
+		err := c.getToken(ctx)
```

Comment:
I'm a little confused. Are we getting a new token for each creation of a stream... and getting the token requires a separate call to etcd ?  I thought we use token stored in the authTokenBundle... as long as we don't get error suggesting `shouldRefreshToken`... 

I know it's not change in this PR, but still if you know, please let me know.

Reply:
I didn't understand why `getToken` needed to be called, but since it was [mentioned in the comment](https://github.com/etcd-io/etcd/blob/7d5151532cf7533198f3f3067152d0fde8efb8f4/client/v3/retry_interceptor.go#L111), I didn't remove it.

But I think that `getToken` is not required on the first run of Interceptor because [getToken is called at client initialization](https://github.com/etcd-io/etcd/blob/7d5151532cf7533198f3f3067152d0fde8efb8f4/client/v3/client.go#L441) and token is stored in the authTokenBundle.

Reply:
Per my understanding `streamClientInterceptor` is just a client side stream Interceptor, but etcd only uses server side streaming (watch stream and lease keep alive), so `streamClientInterceptor` is only executed once on the very beginning of a stream.  Note that for server-side streaming, the client side only calls `SendMsg` once.

So actually the token will never be refreshed during a watch stream, and it's exactly the reason of https://github.com/etcd-io/etcd/issues/12385, which was resolved in https://github.com/etcd-io/etcd/pull/14322. But unfortunately, we rollbacked the fix due to some concern on K8s side.  FYI. https://github.com/etcd-io/etcd/discussions/14992

Reply:
Update: `lease keep alive` should be a bi-directional stream. The client side keeps sending `LeaseKeepAliveRequest`, and server side responds with a `LeaseKeepAliveResponse` each time.

Reply:
Thank you for letting me know. I added comment.

Reply:
@ahrtr @ptabor CI passed and I have nothing to change in this PR :)

Reply:
@wafuwafu13 Please squash the commits and rebase this PR.